+dj/kill_quick.m

```diff
@@ -0,0 +1,36 @@
+% Kill database connections without prompting.
+%   dj.kill_quick() kills MySQL server connections matching 'restriction',
+%   returning the number of terminated connections.
+%
+%   Restrictions are specified as strings and can involve any of the attributes
+%   of information_schema.processlist: ID, USER, HOST, DB, COMMAND, TIME,
+%   STATE, INFO.
+%
+%   Examples:
+%       dj.kill_quick('HOST LIKE "%compute%"') terminates connections from hosts
+%       containing "compute" in their hostname.
+%
+%       dj.kill_quick('TIME > 600') terminates all connections older than 10
+%       minutes.
+
+function nkill = kill_quick(restriction, connection)
+
+    if nargin < 2
+        connection = dj.conn;
+    end
+
+    qstr = 'SELECT * FROM information_schema.processlist WHERE id <> CONNECTION_ID()';
+
+    if nargin && ~isempty(restriction)
+        qstr = sprintf('%s AND (%s)', qstr, restriction);
+    end
+
+    sprintf('%s', qstr)
+
+    res = query(connection, qstr);
+    nkill = 0;
+    for id = double(res.ID)'
```

Comment:
Let's not assume that the keys from the query returned will be upper-cased to be compatible with all MySQL versions. Best to explicitly lower call `struct` keys and then index into it.

Reply:
proceeding here - though wondering if we should push this kind of thing down to queries and standardize on this in both clients, potentially will have alot of useless downcasing loops peppered around the code as time progresses.. will discuss with the group

Reply:
We can push this into queries, though we will still need to index into the result `struct` which would then be lower-cased i.e. still would require this to be 'lowered'.

Reply:
updated - just checked for ID field and selected upper / lower accordingly in this case instead of rebuilding result struct